server.go

```diff
@@ -106,6 +113,10 @@ func NewServer(opt ...ServerOption) *Server {
 	for _, o := range opt {
 		o(&opts)
 	}
+	if opts.codec == nil {
+		// Set the default codec.
+		opts.codec = &protoCodec{}
```

Comment:
Fine with me. My concern was the case in which a user has a complicated big custom codec so that a pointer is needed. Thus probably always passing pointer might be clearer...